### ISSUE TITLE
Suggestion: Double speed key

### ISSUE THREAD

**Issue by GrayFace** [diasurgical/devilutionX#1090]
It would be great to have a key that would toggle between maximum speed and the one you chose in Options. Currently you have to change speed manually each time you enter or leave the town.

**Comment by AJenbo** [comment]
a qol to manage the qol 🤔

**Comment by GrayFace** [comment]
Regular speed setup is more than QOL. I always play at a moderately increased speed. Since there are no numbers for these settings, I have Gamma and Speed at the same place to know where to return speed setting each time I exit the town.

**Comment by FitzRoyX** [comment]
Are you basically asking to fast-foward in dungeons so that backtracking goes faster?

**Comment by GrayFace** [comment]
Yes, and in towns. You can already do it with Speed setting in Options, but it's cumbersome - going through menus and then finding the exact speed value you prefer to play the game with.

**Comment by AJenbo** [comment]
@GrayFace just out of curiosity, what exact value do you normally have it at?

**Comment by GrayFace** [comment]
game speed=26

**Comment by john-parton** [comment]
How's this different from the jog feature from hellfire?

**Comment by AJenbo** [comment]
> How's this different from the jog feature from hellfire?

The whole game speeds, it's active on all levels and doesn't just switch between 1x and 2x. So they actually have more things that put's them apart than things they share, the only common part is that it allows you to move after in town.

From the menu it allows you to increase the speed from 1x to 2.5x, but you can increase it to 50x in the ini.

**Comment by julealgon** [comment]
I'd rather implement a "increase speed" and a "decrease speed" key since that's a more general solution.

As long as your keyboard is configured with a fast repeat rate, going to a "fast enough" speed by holding the key would be done near instantaneously.

**Comment by GrayFace** [comment]
> From the menu it allows you to increase the speed from 1x to 2.5x

Oh, I thought it goes up to 2x, not 2.5x.

> How's this different from the jog feature from hellfire?

I actually learned about jog in Hellfire about a week ago from an article, as I only played original Diablo back in the day. It's certainly nice and would solve the main usecase. As I understand, official support for Hellfire is coming soon, right?

> I'd rather implement a "increase speed" and a "decrease speed" key since that's a more general solution.

It could be if the change step is big enough and doesn't snap to limits. Like, a step of 0.5 (maaybe 0.25). Say, if I have speed of 1.2 it would stay at 1.2 if I press "decrease speed", and would go up to 2.2 if I hold "increase" key or press it twice. This way the preferred speed setting would be preserved and easy to go back to.
